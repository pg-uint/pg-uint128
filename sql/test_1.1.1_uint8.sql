-- Testing cross-types compatibility

SELECT (0::uint8)::json;
SELECT json_build_object('some', 0::uint8);
SELECT ('0'::json)::uint8;
SELECT (0::uint8)::jsonb;
SELECT jsonb_build_object('some', 0::uint8);
SELECT ('0'::jsonb)::uint8;
SELECT ('"abc"'::jsonb)::uint8;
