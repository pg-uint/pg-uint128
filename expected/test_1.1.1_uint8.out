-- Testing cross-types compatibility
SELECT (0::uint8)::json;
 json 
------
 0
(1 row)

SELECT json_build_object('some', 0::uint8);
 json_build_object 
-------------------
 {"some" : 0}
(1 row)

SELECT ('0'::json)::uint8;
 uint8 
-------
 0
(1 row)

SELECT (0::uint8)::jsonb;
 jsonb 
-------
 0
(1 row)

SELECT jsonb_build_object('some', 0::uint8);
 jsonb_build_object 
--------------------
 {"some": 0}
(1 row)

SELECT ('0'::jsonb)::uint8;
 uint8 
-------
 0
(1 row)

SELECT ('"abc"'::jsonb)::uint8;
ERROR:  cannot cast jsonb string to type uint8
